app/classes/observation_report/darwin/images.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+module ObservationReport
+  module Darwin
+    # Darwin Core Observations format.
+    class Images < ObservationReport::CSV
+      attr_accessor :observations, :query, :tables
+
+      self.separator = "\t"
+
+      def initialize(args)
+        super(args)
+        self.observations = args[:observations]
+        initialize_query
+      end
+
+      def initialize_query
+        add_joins
+        add_project
+      end
+
+      def add_joins
+        self.tables = {
+          images: Image.arel_table,
+          images_observations: Arel::Table.new(:images_observations),
+          observations: Observation.arel_table,
+          users: User.arel_table,
+          licenses: License.arel_table
+        }
+        self.query = tables[:images]
+        join_table(:images_observations, :image_id, attribute(:images, :id))
+        io_attribute = attribute(:images_observations, :observation_id)
+        join_table(:observations, :id, io_attribute)
```

Comment:
I always worry whenever we join observations to images.  And I think in this case you don't even need to.  The only column you apparently need from observations is id, which is in images_observations.

Reply:
I understand your worry.  I think it's avoidable in this case as you suggest, but I think the EOL export is going to require it since it will need to connect images to taxa and I'm pretty sure that means going through observations.  Assuming the underlying issue is total memory usage, it might be a time to use the `find_each` approach and limit the queries to 1000 rows or so and then iterate through them.  I expect these will ultimately be happening in a separate job/thread so I'm not terribly worried about how long it takes assuming it doesn't create some sort of table lock.